api/src/main/java/org/apache/iceberg/expressions/BoundTransform.java

```diff
@@ -31,15 +32,17 @@
 public class BoundTransform<S, T> implements BoundTerm<T> {
   private final BoundReference<S> ref;
   private final Transform<S, T> transform;
+  private final Function<S, T> func;
```

Comment:
Same question about serializability of `Function` as `Term` is serializable.

Reply:
Okay, I see that actual classes returned by `bind` are serializable. I wonder whether we should make that part of the contract and have an interface like `SerializableFunction`, similar to `SerializableSupplier`.

Reply:
I think you're right. We should.

Reply:
Done.